java/ql/lib/ext/javax.servlet.model.yml

```
@@ -9,9 +9,15 @@ extensions:
       - ["javax.servlet", "ServletRequest", False, "getParameterNames", "()", "", "ReturnValue", "remote", "manual"]
       - ["javax.servlet", "ServletRequest", False, "getParameterValues", "(String)", "", "ReturnValue", "remote", "manual"]
       - ["javax.servlet", "ServletRequest", False, "getReader", "()", "", "ReturnValue", "remote", "manual"]
-
+      - ["javax.servlet", "ServletRequest", True, "getParameter", "(String)", "", "Parameter[0]", "remote", "ai-manual"]
+      - ["javax.servlet", "ServletRequest", True, "getParameterValues", "(String)", "", "Parameter[0]", "remote", "ai-manual"]
   - addsTo:
       pack: codeql/java-all
       extensible: sinkModel
     data:
       - ["javax.servlet", "ServletContext", True, "getResourceAsStream", "(String)", "", "Argument[0]", "path-injection", "ai-manual"]
+  - addsTo:
+      pack: codeql/java-all
+      extensible: summaryModel
+    data:
+      - ["javax.servlet", "ServletRequest", True, "getRealPath", "(String)", "", "Parameter[0]", ReturnValue, "taint", "ai-manual"]
```

This should be `Argument[0]`.

```suggestion
      - ["javax.servlet", "ServletRequest", True, "getRealPath", "(String)", "", "Argument[0]", ReturnValue, "taint", "ai-manual"]
```

---

But what we usually do is making the return value of such calls a source directly — I don't think we have models that taint the `ServletRequest` directly.

---

> But what we usually do is making the return value of such calls a source directly

Oh, interesting! That was, in fact, the prediction we saw during triage.

---

> Oh, interesting! That was, in fact, the prediction we saw during triage.

It really is interesting! I think in this case, I'd like to accept the originally proposed model 🥳 I'll update the PR.

---

Oh, wait 🤦 I'm really sorry, I had read this method wrong. I thought that it returned the real path of the request, but it actually receives the controllable part as a parameter, not from the request itself.

So this is indeed a summary, not a source, because the external user can't really control anything from the return value (the `host:port/contextPath` part is predefined, and the `/index.html` part is actually the `path` argument).

Sorry for the confusion, I'll fix it myself and merge.